src/rust/src/lazyframe/general.rs

```diff
@@ -273,9 +270,149 @@ impl PlRLazyFrame {
+            .collect::<Vec<PathBuf>>();
+        let encoding = <Wrap<CsvEncoding>>::try_from(encoding)?.0;
+        let skip_rows = <Wrap<usize>>::try_from(skip_rows)?.0;
+        let skip_rows_after_header = <Wrap<usize>>::try_from(skip_rows_after_header)?.0;
+        let infer_schema_length: Option<usize> = match infer_schema_length {
+            Some(x) => Some(<Wrap<usize>>::try_from(x)?.0),
+            None => None,
+        };
+        let row_index_offset = <Wrap<u32>>::try_from(row_index_offset)?.0;
+        let n_rows: Option<usize> = match n_rows {
+            Some(x) => Some(<Wrap<usize>>::try_from(x)?.0),
+            None => None,
+        };
+        let null_values: Option<NullValues> = match null_values {
+            Some(x) => Some(<Wrap<NullValues>>::try_from(x)?.0),
+            None => None,
+        };
+        let retries = <Wrap<usize>>::try_from(retries)?.0;
+        let file_cache_ttl: Option<u64> = match file_cache_ttl {
+            Some(x) => Some(<Wrap<u64>>::try_from(x)?.0),
+            None => None,
+        };
 
-        let first_path = source.first().unwrap().clone().into();
+        let quote_char = quote_char
+            .map(|s| {
+                s.as_bytes()
+                    .first()
+                    .ok_or_else(|| polars_err!(InvalidOperation: "`quote_char` cannot be empty"))
+            })
+            .transpose()
+            .map_err(RPolarsErr::from)?
+            .copied();
+        let separator = separator
+            .as_bytes()
+            .first()
+            .ok_or_else(|| polars_err!(InvalidOperation: "`separator` cannot be empty"))
+            .copied()
+            .map_err(RPolarsErr::from)?;
+        let eol_char = eol_char
+            .as_bytes()
+            .first()
+            .ok_or_else(|| polars_err!(InvalidOperation: "`eol_char` cannot be empty"))
+            .copied()
+            .map_err(RPolarsErr::from)?;
 
+        let row_index: Option<RowIndex> = match row_index_name {
+            Some(x) => Some(RowIndex {
+                name: x.into(),
+                offset: row_index_offset,
+            }),
+            None => None,
+        };
+
+        let overwrite_dtype: Option<Vec<(&str, DataType)>> = match overwrite_dtype {
+            Some(x) => {
+                let list_len = x.len();
+                let mut vec = Vec::with_capacity(x.len());
+                let names_list = x.iter().map(|x| x.0).collect::<Vec<&str>>();
+                let vec_dt = <Wrap<Vec<DataType>>>::try_from(x)?.0;
+                for i in 0..list_len {
+                    vec.push((names_list[i], vec_dt[i].clone()));
+                }
+                Some(vec)
+            }
+            None => None,
+        };
+        let overwrite_dtype = overwrite_dtype.map(|overwrite_dtype| {
+            overwrite_dtype
+                .into_iter()
+                .map(|(name, dtype)| Field::new((&*name).into(), dtype))
+                .collect::<Schema>()
+        });
```

Comment:
`<Wrap<Vec<Field>>>::try_from(overwrite_dtype)` ?

Reply:
I don't understand what you want to replace, can you apply your change?